v0/destinations/google_ads_classic/transform.js

```diff
@@ -0,0 +1,187 @@
+  const reservedEventNames = [
+    "first_open",
+    "session_start",
+    "in_app_purchase",
+    "view_item_list",
+    "view_item",
+    "view_search_results",
+    "add_to_cart",
+    "ecommerce_purchase",
+    "custom"
+  ];
+  const eventName = get(message, "event");
+  /**  check if the event name is one of the default event types.
+   * This field must not contain any of the values reserved for app_event_type.
+   * If a reserved event name is used, the API will return an APP_EVENT_NAME_RESERVED_VALUE error.
+   */
+
+  if (reservedEventNames.indexOf(eventName) >= 0) {
+    throw new Error(
+      "Event name is a reserved event name in Google Ads. Aborting Message"
+    );
+  }
+  return eventName;
+}
+/**
+ *
+ * @param {Rudder Payload} message
+ * To set app_event_type
+ * The name of the app event that occurred. This field is an enumeration, and will only accept valued predefined by google ads.
+ * https://developers.google.com/app-conversion-tracking/api/request-response-specs#conversion_tracking_request
+ *
+ */
+function setEventType(message) {
+  const eventName = get(message, "event");
+  switch (eventName.toLowerCase()) {
+    case "application installed":
+      return "first_open";
+    case "application opened":
+      return "session_start";
+    case "order completed":
+      // Use in_app_purchase for purchases made through the native app store; use ecommerce_purchase for all other purchases.
+      if (
+        get(message, "properties.affiliation") === "App Store" ||
+        get(message, "properties.affiliation") === "Google Store"
+      ) {
+        return "in_app_purchase";
+      }
+      return "ecommerce_purchase";
+    case "product list viewed":
+      return "view_item_list";
+    case "product added":
+      return "add_to_cart";
+    case "poducts searched":
+      return "view_search_results";
+    case "product viewed":
+      return "view_item";
+    default:
+      return "custom";
+  }
+}
+function responseBuilderSimple(message, category, destination) {
+  const {
+    devToken,
+    linkId,
+    version,
+    lat,
+    fallbackIDFAZero
+  } = destination.Config;
+  // if fall back to value is false and rdid is not present the message will not be accepted by google ads so it will be aborted by rudder.
+  if (!fallbackIDFAZero && !get(message, "context.device.advertisingId")) {
+    throw new Error("No advertisingId set. Aborting message");
+  }
+  const payload = constructPayload(message, category);
+  if (payload) {
+    payload.dev_token = devToken;
+    payload.link_id = linkId;
+    payload.app_event_type = setEventType(message);
+    // This field is required if app_event_type is custom.
+    if (payload.app_event_type === "custom") {
+      payload.app_event_name = setEventName(message);
+    }
+    /**
+     * Limit-ad-tracking status for the device.
+     * 0: User has not chosen to limit ad tracking.
+     * 1: User has chosen to limit ad tracking.
+     */
+    payload.lat = lat ? 1 : 0;
```

Comment:
Rename to `limitAdTracking` for readibility

Reply:
done